### PR TITLE
FreeRTOS Kernel Spelling Update

### PR DESCRIPTION
**Pull request by lundinc2**
FreeRTOS Kernel Spelling Update

Description
-----------
* Added spell check to kernel repository.
* Fixed small spelling errors in various kernel source files.

Note: Only kernel files are checked for spelling, and portable files are ignored.

Test Steps
-----------

Related Issue
-----------

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by lundinc2** [comment]
TODO: Update the README with documentation for this change.

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+          sudo apt-get install spell
+          sudo apt-get install util-linux
+      - name: Check spelling
+        run: |
+          PATH=$PATH:main/tools/spell
+          ls kernel
+          # Make sure that the portable directory is not included in the spellcheck.
+          sed -i 's/find $DIRNAME/find $DIRNAME -not -path '*portable*'/g' main/tools/spell/find-unknown-comment-words
+          find-unknown-comment-words --directory kernel/
+          if [ "$?" = "0" ]; then
```
Comment (alfred2g) [inline .github/workflows/ci.yml:33]: "==" ?

File: README.md [hunk 2]
```diff
@@ -33,3 +33,7 @@ See the readme file in the ```./portable``` directory for more information.
 ### Code Formatting
 FreeRTOS files are formatted using the "uncrustify" tool. The configuration file used by uncrustify can be found in the [FreeRTOS/FreeRTOS repository](https://github.com/FreeRTOS/FreeRTOS/blob/master/tools/uncrustify.cfg). 
 
+### Spelling
+*lexicon.txt* contains words that are not traditionally found in an English dictionary. It is used by the spellchecker to verify the various jargon, variable names, and other odd words used in the FreeRTOS code base. If your pull request fails to pass the spelling check fix the error. If it is a variable name or the like that IS spelled correctly, but unknown to the lexicon, then add the word to *lexicon.txt*.
```
Comment (alfred2g) [inline README.md:37]: If your pull request fails to pass the spelling check fix the error.  odd sentence

Comment (lundinc2) [reply README.md:37]: Hmm true, must have been tired :) I will fix this

**Review by cobusve — APPROVED** [review]

**Pull request merged**